### ISSUE TITLE
[Issue]: /var/lib/dkms/amdgpu isn't removed on uninstall and caused dkms.conf error

### ISSUE THREAD

**Issue by asvishnyakov** [ROCm/ROCm#4197]
### Problem Description

amgpu-install 6.2.4 and later causes

```
Error! Could not locate dkms.conf file.
File: /var/lib/dkms/amdgpu/6.8.5-2041575.24.04/source/dkms.conf does not exist.
```

error when trying to install dkms

### Operating System

Ubuntu 24.04.1 LTS (Noble Numbat)

### CPU

N/A

### GPU

N/A

### ROCm Version

ROCm 6.3.1

### Steps to Reproduce

1. Install Ubuntu 24.04.1
2. Install amggpu 6.2.4 or later (this issue was originally about 6.3.1, but I learned only 6.2.* compatible with Ubuntu 24.04.1) following https://rocm.docs.amd.com/projects/install-on-linux/en/latest/install/amdgpu-install.html
3. Run `sudo amdgpu-install --usecase=dkms`

**Comment by ppanchad-amd** [comment]
Hi @asvishnyakov. Internal ticket has been created to investigate your issue. Thanks!

**Comment by asvishnyakov** [comment]
Oh ok, it looks like I should still use 6.2.* with 24.04.1. That's strange, because 24.04.2 isn't released yet

**Comment by asvishnyakov** [comment]
@ppanchad-amd I reopen this issue as downgrading to 6.2.4 changed nothing, even with all cleanups

**Comment by asvishnyakov** [comment]
This looks like duplicate of #3608, but instructions specified there didn't help me, `/var/lib/dkms/amdgpu` was left after uninstallation even with all uninstall/purge/autoremove commands and flags. I was able to install dkms of 6.2* version for my Ubuntu 24.04.1 successfully only after I manually removed full folder with it content.

**Comment by lucbruni-amd** [comment]
Hi @asvishnyakov, thank you for creating this issue and reopening after further investigation. An internal ticket has been created for this and we are investigating.

**Comment by lucbruni-amd** [comment]
Hi @asvishnyakov, I can indeed see that running the following: 
`sudo apt purge amdgpu-install
sudo apt autoremove`
does not remove the corresponding version entry like such: `/var/lib/dkms/amdgpu/<entry>`, but does remove everything inside the entry.

I am not able to reproduce on Ubuntu 24.04.1 LTS (Noble Numbat).
Could you kindly provide the steps you took to:
1. Install the initial version you later uninstalled
2. Uninstall said version
3. Attempt to upgrade to a later but different version, causing the error you encountered

Thank you!

**Comment by lucbruni-amd** [comment]
Hi @asvishnyakov, are you still encountering this issue? If so, please let me know the steps mentioned in my previous comment. Thanks!

**Comment by lucbruni-amd** [comment]
Closing this issue due to inactivity. If the issue persists, feel free to open another ticket with detailed steps taken to reproduce the issue. Thanks!
